src/PhpSpreadsheet/Worksheet/Worksheet.php

```diff
@@ -2007,6 +2017,74 @@ public function removeAutoFilter(): self
         return $this;
     }
 
+    /**
+     * Get collection of Tables.
+     *
+     * @return ArrayObject<int, Table>
+     */
+    public function getTableCollection()
+    {
+        return $this->tableCollection;
+    }
+
+    /**
+     * Add Table.
+     *
+     * @return $this
+     */
+    public function addTable(Table $table): self
+    {
+        $table->setWorksheet($this);
+        $this->tableCollection[] = $table;
+
+        return $this;
+    }
+
+    /**
+     * Add Table Range by using numeric cell coordinates.
```

Comment:
Note that I'm trying to deprecate the "ByColumnAndRow" methods in Worksheet

Reply:
Should I need to remove the method? Or replace it with `addTableByCellAddress`?

Reply:
My thought is to remove the method completely. And it would be better to add support for `string|CellAddress` while instantiating the table.

Reply:
That would be the best solution I think. We've never been consistent in our use of the "ByColumnAndRow" methods, so I'm trying to make things consistent for v2.0 by removing those methods, and deprecating those existing uses now. Using string|array<int>|CellRange allows a lot more flexibility as well; if you just copy the way the AutoFilter has been changed, it should work well.

Reply:
``` php
$cellRange = Functions::trimSheetFromCellReference($this->validateCellRange($autoFilterOrRange));
```

I found this in the `setAutoFilter` method of `Worksheet.php`. And I need your help to use it in Table class. I have a few questions. 
- Is there any helper function similar to `validateCellRange` (outside worksheet.php)?
- Why `validateCellRange` visibility is protected and is not static (like `extractSheetTitle`) ?

Reply:
There's nothing similar to `validateCellRange()` outside of the Worksheet object because it was introduced as validation for methods that existed only in the Worksheet class and not elsewhere.
It's not static, because the Worksheet object must have been instantiated already to use the non-static methods that reference it; and also (potentially in the future) so that it can access the Worksheet title for additional validation.
It's protected rather than public because there was no need to call it from outside of the Worksheet class; and any public method becomes accessible to user code, which isn't necessarily a good thing.

It isn't yet in any released code because it's only been created since the last release; so if you want to call it from your Table class, I can extract it into a separate `Worksheet/Validations` class; and I can make those methods static and public. Just let me know if you want me to do that.

Reply:
Thank you for clarifying that. 

Please extract it into a separate `Worksheet/Validations` class for me. Thank you once again.

Reply:
I've extracted those validation methods into a separate class, and made them all public static. I've also applied those validations to the cell range checking in AutoFilter (though only minimally, that code could be cleaned up a bit more)... that should also help provide an example for your own usage.

Reply:
Implemented in 534cbc04c040079c26c8855670ba8ed94566c176

And I have noticed that the `$worksheet` is nullable, but the default argument value `null` is missing in the [CellAddress constructor](https://github.com/PHPOffice/PhpSpreadsheet/blob/master/src/PhpSpreadsheet/Cell/CellAddress.php#L35).

```diff
-public function __construct(string $cellAddress, ?Worksheet $worksheet)
+public function __construct(string $cellAddress, ?Worksheet $worksheet = null)
```

Reply:
Thanks, I'll fix that